tests/test_tts.py

```diff
@@ -97,3 +136,29 @@ def test_Tiro_speech_synthesis():
     assert path.is_file(), "Expected audio file to exist"
     assert path.stat().st_size > _MIN_AUDIO_SIZE, "Expected longer audio data"
     path.unlink()
+
+
+@patch.dict(SERVICES, {"mock_service": MagicMock()})
+@patch.dict(VOICES, {"Dora": {"service": "mock_service"}})
+def test_keys_override_in_tts_to_file():
+    """Test if keys_override is correctly passed into service.text_to_speech."""
+    _TEXT = "Test"
+    SERVICES["mock_service"].audio_formats = ["mp3"]
+    keys_override = Keys(
+        aws=AWSPollyKey(
+            aws_access_key_id="test",
+            aws_secret_access_key="test",
+            region_name="test",
+        )
+    )
+    tts_to_file(
+        _TEXT,
+        TTSOptions(text_format=TextFormats.TEXT, audio_format="mp3", voice="Dora"),
+        transcribe=False,
+        keys_override=keys_override,
+    )
+    SERVICES["mock_service"].text_to_speech.assert_called_once_with(
+        _TEXT,
+        TTSOptions(text_format=TextFormats.TEXT, audio_format="mp3", voice="Dora"),
+        keys_override,
+    )
```

Comment:
If the `__eq__` and `__hash__` methods are implemented for this to work, does it work simply passing in the same instance? i.e. `opt = TTSOptions(text_format=...)` and passing into both functions? Otherwise, wonderful tests :)

Reply:
Hmm not sure I understand - the override keys exist outside of the TTSOptions so we are just passing in the same instance in both cases as an additional parameter to `tts_to_file` and `text_to_speech`.

Reply:
My idea was something like:
```py
opts = TTSOptions(text_format=TextFormats.TEXT, audio_format="mp3", voice="Dora")
tts_to_file(
    _TEXT,
    opts,
    transcribe=False,
    keys_override=keys_override,
)
SERVICES["mock_service"].text_to_speech.assert_called_once_with(
    _TEXT,
    opts,
    keys_override,
)
```
Do you know if that could work? As it is then the same instance in the comparison taking place within `assert_called_once_with`.
If this works, we could remove the `__eq__` and `__hash__` methods (simply _slightly_ less code to take care of in the future, but this is otherwise very minor).